cache.go

```diff
@@ -138,7 +138,11 @@ func (c *cache) create(t reflect.Type, info *structInfo) *structInfo {
 				ft = ft.Elem()
 			}
 			if ft.Kind() == reflect.Struct {
+				bef := len(info.fields)
 				c.create(ft, info)
+				for _, fi := range info.fields[bef:len(info.fields)] {
```

Comment:
Can you add a code comment as to what is happening here and why `fi.required` is being set to false?